diag_manager/fms_diag_reduction_methods.F90

```diff
@@ -0,0 +1,695 @@
+        ! Update the buffer with the current minimum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) <&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        else
+        ! Update the buffer with the current maximum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) >&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        end if
+      class default
+        call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum type mismatch")
+      end select
+    class default
+      call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum unsupported field data type")
+    end select
+  end subroutine update_scalar_extremum
+
+  !> @brief Updates a chunk of buffer
+  subroutine update_array_extremum(flag, field_data, buffer, mask, sample, recon_bounds, reduced_k_range)
+    integer :: flag !< 0 for minimum; 1 for extremum
+    class(*), intent(in) :: field_data(:,:,:,:) !< Field data
+    class(*), intent(inout) :: buffer(:,:,:,:,:) !< Remapped output buffer
+    logical, intent(in) :: mask(:,:,:,:) !< Updated mask
+    integer, intent(in) :: sample !< diurnal sample index
+    type(fmsDiagBoundsHalos_type), intent(in) :: recon_bounds !< Object to hold starting and ending indices
+                                                              !! in the I, J, and K dimensions; also holds
+                                                              !! halo sizes in the I, and J dimensions
+    logical, intent(in) :: reduced_k_range !< Flag indicating if the range in the K dimension is present
+
+    integer :: is, js, ks !< Starting indices in the I, J, and K dimensions
+    integer :: ie, je, ke !< Ending indices in the I, J, and K dimensions
+    integer :: hi, hj !< Halo sizes in the I, and J dimensions
+    integer :: f1, f2 !< Updated starting and ending indices in the I dimension
+    integer :: f3, f4 !< Updated starting and ending indices in the J dimension
+
+    !> Unpack bounds (/is, js, ks, ie, je, ke, hi, f1, f2, hj, f3, f4/)
+    is = recon_bounds%get_bounds3D()%get_imin()
+    js = recon_bounds%get_bounds3D()%get_jmin()
+    ks = recon_bounds%get_bounds3D()%get_kmin()
+    ie = recon_bounds%get_bounds3D()%get_imax()
+    je = recon_bounds%get_bounds3D()%get_jmax()
+    ke = recon_bounds%get_bounds3D()%get_kmax()
+    hi = recon_bounds%get_hi()
+    f1 = recon_bounds%get_fis()
+    f2 = recon_bounds%get_fie()
+    hj = recon_bounds%get_hj()
+    f3 = recon_bounds%get_fjs()
+    f4 = recon_bounds%get_fje()
+
+    ! Select proper type and update the buffer
+    select type (field_data)
+    type is (real(kind=r4_kind))
+      select type (buffer)
+      type is (real(kind=r4_kind))
+        if (flag .eq. 0) then
```

Comment:
Please label some of these larger `IF` statements in this routine.  It would make it much easier to follow if they were labeled.

Reply:
done.